new-cdc/eventuate-local-java-cdc-connector-mysql-binlog/src/main/java/io/eventuate/local/mysql/binlog/MySqlBinaryLogClient.java

```diff
@@ -118,20 +134,14 @@ protected void leaderStart() {
         case TABLE_MAP: {
           TableMapEventData tableMapEvent = event.getData();
 
-          SchemaAndTable schemaAndTable = new SchemaAndTable(tableMapEvent.getDatabase(), tableMapEvent.getTable());
-
-//          if (!cdcMonitoringTableId.isPresent() &&
-//                  schemaAndTable.equals(new SchemaAndTable(new EventuateSchema().getEventuateDatabaseSchema(), "cdc_monitoring"))) {
-//            cdcMonitoringTableId = Optional.of(tableMapEvent.getTableId());
-//            break;
-//          }
+          cdcMonitoringTableId = Optional.empty();
+          tableMapEventByTableId.clear();
```

Comment:
Hmmm.... are you sure you should clear it.
This appears to suggest that its a map with at most one entry?

Reply:
From the docs, TABLE_MAP event precedes each row operation event.
So required id is always present in map when write/update event is received. So it should work.
In current case it solves the issue with changed ids when mysql is restarted.
I am not sure that there is better solution.